test/widgets/message_list_test.dart

```diff
@@ -610,13 +610,28 @@ void main() {
 
   group('MarkAsReadWidget', () {
     bool isMarkAsReadButtonVisible(WidgetTester tester) {
-      // Zero height elements on the edge of a scrolling viewport
-      // are treated as invisible for hit-testing, see
-      // [SliverMultiBoxAdaptorElement.debugVisitOnstageChildren].
-      // Set `skipOffstage: false` here to safely target the
-      // [MarkAsReadWidget] even when it is inactive.
-      return tester.getSize(
-        find.byType(MarkAsReadWidget, skipOffstage: false)).height > 0;
+      // Locate the [MarkAsReadWidget] and check its state
+      // to determine if it is visible.
+      var markAsReadWidget = find.byType(MarkAsReadWidget);
+
+      // The [MarkAsReadWidget] is a [IgnorePointer] wrapped in a [AnimatedOpacity].
+      // We need to check the state of both to determine if the widget is visible.
+      // The [IgnorePointer] is visible when `ignoring` is false and the [AnimatedOpacity]
+      // is visible when `opacity` is 1.0.
+      final ignorePointerFinder = find.descendant(
+        of: markAsReadWidget,
+        matching: find.byType(IgnorePointer),
+      );
+
+      final animatedOpacityFinder = find.descendant(
+        of: markAsReadWidget,
+        matching: find.byType(AnimatedOpacity),
+      );
+
+      final IgnorePointer ignorePointer = tester.widget(ignorePointerFinder);
+      final AnimatedOpacity animatedOpacity = tester.widget(animatedOpacityFinder);
+
+      return !ignorePointer.ignoring && animatedOpacity.opacity == 1.0;
     }
 
     testWidgets('from read to unread', (WidgetTester tester) async {
```

Comment:
In addition to updating the existing tests, this change should add a new test: we want to check that the messages don't shift position when marking as read.

Reply:
I have added a new test that addresses this. It checks if the vertical position of the first message item is the same before and after the "Mark all messages as read" button is pressed.